pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -702,6 +702,11 @@ protected static PartitionedTopicMetadata fetchPartitionedTopicMetadata(PulsarSe
             return pulsar.getBrokerService().fetchPartitionedTopicMetadataAsync(topicName).get();
         } catch (Exception e) {
             if (e.getCause() instanceof RestException) {
+                // Since CompletableFuture wrappers exception, so we will lost the Status
+                // rebuild the Status if exception message contains "Topic not exist."
+                if (e.getCause().getMessage().contains("Topic not exist.")) {
+                    throw new org.apache.pulsar.broker.web.RestException(Response.Status.NOT_FOUND, "Topic not exist.");
+                }
```

Comment:
Could you please clarify why `throw (RestException) e.getCause()` does not work here? If e.getCause is instance of RestException, I think we can throw it directly. I might be missing something here.

Reply:
I'll check this PR and push later.

Reply:
Thanks @aloyszhang